src/Octoshift/AdoApi.cs

```diff
@@ -262,8 +264,25 @@ public virtual async Task<string> GetTeamProjectId(string org, string teamProjec
         public virtual async Task<string> GetRepoId(string org, string teamProject, string repo)
         {
             var url = $"https://dev.azure.com/{org}/{teamProject}/_apis/git/repositories/{repo}?api-version=4.1";
-            var response = await _client.GetAsync(url);
-            return (string)JObject.Parse(response)["id"];
+            try
+            {
+                var response = await _client.GetAsync(url);
+                return (string)JObject.Parse(response)["id"];
+            }
+            catch (HttpRequestException ex)
+            {
+                if (ex.StatusCode == HttpStatusCode.NotFound)
+                {
+                    // The repo may be disabled, can still get the ID by getting it from the repo list
+                    url = $"https://dev.azure.com/{org}/{teamProject}/_apis/git/repositories?api-version=4.1";
+
+                    var response = await _client.GetWithPagingAsync(url);
+
+                    return (string)response.Single(x => ((string)x["name"]) == repo)["id"];
+                }
+
+                throw;
+            }
```

Comment:
if we're only going to check for `NotFound` to remove indentation we can use `when`:
```suggestion
            catch (HttpRequestException ex) when (ex.StatusCode == HttpStatusCode.NotFound)
            {
                // The repo may be disabled, can still get the ID by getting it from the repo list
                url = $"https://dev.azure.com/{org}/{teamProject}/_apis/git/repositories?api-version=4.1";

                var response = await _client.GetWithPagingAsync(url);

                return (string)response.Single(x => ((string)x["name"]) == repo)["id"];
            }
```

Reply:
first time I've seen that syntax, I like!